test/test_suite.py

```diff
@@ -198,3 +213,191 @@ class TableDDLTest(_TableDDLTest):
+
+        row = config.db.execute(select([date_table.c.date_data])).first()
+
+        compare = self.compare or self.data
+        eq_(row, (compare,))
+        assert isinstance(row[0], type(compare))
+
+    def test_null(self):
+        """
+        SPANNER OVERRIDE:
+
+        Cloud Spanner supports tables with empty primary key, but
+        only single one row can be inserted into such a table -
+        following insertions will fail with `400 id must not be NULL in table date_table`.
+        Overriding the tests and add a manual primary key value to avoid the same failures.
+        """
+        date_table = self.tables.date_table
+
+        config.db.execute(date_table.insert(), {"id": 1, "date_data": None})
+
+        row = config.db.execute(select([date_table.c.date_data])).first()
+        eq_(row, (None,))
+
+    @requires.standalone_null_binds_whereclause
+    def test_null_bound_comparison(self):
+        """
+        SPANNER OVERRIDE:
+
+        Cloud Spanner supports tables with empty primary key, but
+        only single one row can be inserted into such a table -
+        following insertions will fail with `400 id must not be NULL in table date_table`.
+        Overriding the tests and add a manual primary key value to avoid the same failures.
+        """
+
+        # this test is based on an Oracle issue observed in #4886.
+        # passing NULL for an expression that needs to be interpreted as
+        # a certain type, does the DBAPI have the info it needs to do this.
+        date_table = self.tables.date_table
+        with config.db.connect() as conn:
+            result = conn.execute(
+                date_table.insert(), {"id": 1, "date_data": self.data}
+            )
+            id_ = result.inserted_primary_key[0]
+            stmt = select([date_table.c.id]).where(
+                case(
+                    [
+                        (
+                            bindparam("foo", type_=self.datatype) != None,
+                            bindparam("foo", type_=self.datatype),
+                        )
+                    ],
+                    else_=date_table.c.date_data,
+                )
+                == date_table.c.date_data
+            )
+
+            row = conn.execute(stmt, {"foo": None}).first()
+            eq_(row[0], id_)
+
+
+class DateTimeMicrosecondsTest(_DateTimeMicrosecondsTest):
+    @classmethod
+    def define_tables(cls, metadata):
+        """
+        SPANNER OVERRIDE:
+
+        Spanner is not able cleanup data and drop the table correctly,
+        table was already exists after related tests finished, so it doesn't
+        create a new table and when started tests for other data type  following
+        insertions will fail with `400 Invalid value for column date_data in
+        table date_table: Expected DATE`.
+        Overriding the tests to create a new table for test to avoid the same failures.
+        """
```

Comment:
Minor nit:
```
        Spanner is not able cleanup data and drop the table correctly,
        table already exists after related tests finished, so it doesn't
        create a new table and insertions for tests for other data types
        will fail with `400 Invalid value for column date_data in
        table date_table: Expected DATE`.
        Overriding the tests to create a new table for tests to avoid the same failures.
```